common/appscale/common/deployment_config.py

```diff
@@ -29,24 +27,56 @@ class ConfigStates(object):
   LOADED = 'loaded'
 
 
+class DeploymentConfigSection(object):
+  """ Keeps track of a section of configuration data. """
+  def __init__(self, zk_client, section):
+    """ Creates a new DeploymentConfigSection.
+
+    Args:
+      zk_client: A KazooClient.
+      section: A string specifying a configuration section name.
+    """
+    self.logger = logging.getLogger(self.__class__.__name__)
+    self.section_name = section
+    self.data = {}
+
+    section_node = '/appscale/config/{}'.format(section)
+    self.watch = zk_client.DataWatch(section_node, self._update_section)
+
+  def stop(self):
+    """ Stops the DataWatch on the configuration section. """
+    self.watch._stopped = True
```

Comment:
I see that there is no obvious way to stop watcher using public API, but relying on non-public attribute still doesn't look good.
My alternative also relies on implementation of DataWatch, but it doesn't use non-public attributes, so I think you can take a look at this:
zk_client.remove_listener - https://github.com/python-zk/kazoo/blob/master/kazoo/client.py#L425
which is opposite to add_listener that is used in DataWatch - https://github.com/python-zk/kazoo/blob/master/kazoo/recipe/watchers.py#L150